### ISSUE TITLE
Abstract out the server and build system

### ISSUE THREAD

**Issue by michaelsharpe** [myplanet/graphql-server#2]
Like NextJS, our implementation should not just be a seed, but an opinionated build system.  This means it can be included as a package in other projects, and enforces a certain structure to reading modules, permissions, etc.

The folder structure we be enforced by auto-loading of files in their proper places.  Individual modules could also be abstracted out into their own packages, such as one for user auth, etc.

Basically, we want this to be more of a package that can be included and continually updated then a starter seed that is forked and modified.
